src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/GeneratedComInterfaceAttributeAnalyzer.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+#nullable enable
+namespace Microsoft.Interop
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class GeneratedComInterfaceAttributeAnalyzer : DiagnosticAnalyzer
+    {
+        private const string _generatedComInterfaceAttributeName = "GeneratedComInterfaceAttribute";
+        private const string _generatedComInterfaceAttributeNamespace = "System.Runtime.InteropServices.Marshalling";
+
+        private const string _interfaceTypeAttributeName = "InterfaceTypeAttribute";
+        private const string _interfaceTypeAttributeNamespace = "System.Runtime.InteropServices";
+
```

Comment:
Might make sense to start a file with constants like these. I imagine we'll need them elsewhere.